### PR TITLE
Ransei

### PR DESCRIPTION
**Pull request by ghost**
Im 95% sure i implemented this wrong and would like guidance pls ty

### CODE REVIEW

File: data/mods/ssb/abilities.js [hunk 1]
```diff
@@ -815,6 +815,38 @@ let BattleAbilities = {
...
+		onAfterDamage(damage, target, source, move) {
+			if (target.getMoveHitData(move).typeMod > 0) {
+				if (!target.heavilydamaged) {
+					this.add('-message', `${target.name}'s attack was reduced after that super effective attack!`);
+					target.heavilydamaged = true;
+				}
+				this.add(`c|@Ransei|Yo really? Why do you keep hitting me with super effective moves?`);
+			}
+		},
+		onModifyAtk(atk, pokemon) {
```
Comment (ghost) [inline data/mods/ssb/abilities.js:839]: I am not sure if i did the correctly due to how the abilities interact with each other

File: data/mods/ssb/scripts.js [hunk 2]
```diff
@@ -292,6 +292,13 @@ let BattleScripts = {
 		this.runEvent('TerrainStart', source, source, status);
 		return true;
 	},
+	modifyDamage(baseDamage, pokemon, target, move, suppressMessages) {
```
Comment (ghost) [inline data/mods/ssb/scripts.js:295]: im not sure if this was implemented properly >.>

File: data/mods/ssb/abilities.js (hunk 1 above)
Comment (HoeenCoder) [reply data/mods/ssb/abilities.js:839]: I think this is ok.

File: data/mods/ssb/abilities.js [hunk 3]
```diff
@@ -815,6 +815,38 @@ let BattleAbilities = {
...
+		name: "Superguarda",
+		isNonstandard: "Custom",
+		onDamage(damage, target, source, effect) {
+			if (effect.effectType !== 'Move') {
+				return false;
+			}
+		},
+		onAfterDamage(damage, target, source, move) {
+			if (target.getMoveHitData(move).typeMod > 0) {
+				if (!target.heavilydamaged) {
```
Comment (HoeenCoder) [inline data/mods/ssb/abilities.js]: Pretty sure theres a `pokmeon.m` value we can make use of (used for OMs) to add values. I cant look into it now but will later if you are too busy.

File: data/mods/ssb/scripts.js (hunk 2 above)
Comment (HoeenCoder) [reply data/mods/ssb/scripts.js:295]: Did you copy/paste the implementation for this from `data/scripts.js` and make the change on L296? If so this should be ok.

**Pull request merged**